packages/server/src/manifest-builder.ts

```diff
@@ -21,14 +21,31 @@ interface ManifestBuilderOptions {
   distDir: string;
 };
 
+function* updateSourceMapURL(filePath: string, sourcemapName: string){
+  let { size } = fs.statSync(filePath);
+  let readStream = fs.createReadStream(filePath, {start: size - 16});
+  readStream.on('data', currentURL => {
```

Comment:
When working with effection, anything asynchronous needs to be an operation. The reasons are many, but the most important ones are:

1. If an error occurs while doing the url substitution, then we need to raise an error in the `updateSourceMapURL` operation.
2. The `updateSourceMapURL` operation, like _any_ effection operation, can be halted at any time. As we have it here, the `readStream.on` callback will continue to fire, even if the caller has gone out of scope for whatever reason.
3. If either 1 or 2 happens, we need to unregister the `on('data')` callback, otherwise it will be firing into dead code.

The same applies to the call to `truncate()`

You can use the `on` operation from `@effection/events`, and the `fs.promises` api to use operations everywhere:

```ts
import { on } from '@effection/events';

function* updateSourceMapURL(filePath: string, sourcemapName: string){
   let { size } = fs.statSync(filePath);
   let readStream = fs.createReadStream(filePath, {start: size - 16});
   let currentURL = yield on(readStream, 'data');
   if (currentURL === '/manifest.js.map') {
     yield fs.promises.truncate(filePath, size - 16);
     fs.appendFileSync(filePath, sourcemapName);
   }
}

Reply:
Oh I see. That makes sense. 👍

Reply:
When I do `let currentURL = yield on(readStream, 'data')`, I get back `ChainableSubscription`. Do I need to import and use `Subscribable.from()` to get the value?

Reply:
Does this look okay?
```
function* updateSourceMapURL(filePath: string, sourcemapName: string){
  let { size } = fs.statSync(filePath);
  let readStream = fs.createReadStream(filePath, {start: size - 16});
  let onRead = yield on(readStream, 'data');
  let currentURL = yield onRead.next();
  
  if( currentURL.value.toString('utf-8') == '/manifest.js.map'){
    yield truncate(filePath, size - 16);
    fs.appendFileSync(filePath, sourcemapName);
  };
}
```